### PR TITLE
chore(deps): update typescript-eslint monorepo to v6

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: demo/src/lib/stackblitz/angular/tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,6 @@
 		"skipLibCheck": true,
 		"esModuleInterop": true,
 		"resolveJsonModule": true,
-		"moduleResolution": "node",
```
Comment (divdavem) [inline demo/src/lib/stackblitz/angular/tsconfig.json:11]: This change breaks angular samples in stackblitz.

File: eslint-plugin/vitest.config.ts [hunk 2]
```diff
@@ -6,5 +6,6 @@ export default defineConfig({
 		coverage: {
 			reporter: ['text', 'json', 'html'],
 		},
+		globals: true,
```
Comment (divdavem) [inline eslint-plugin/vitest.config.ts:9]: I think it is better not to expose globals, and instead to explicitly link the functions needed by the rule tester in `svelte-check-props.spec.ts`:
```ts
import {afterAll, describe, test} from 'vitest';
// ...
import {RuleTester} from '@typescript-eslint/rule-tester';

RuleTester.describe = describe;
RuleTester.it = test;
RuleTester.afterAll = afterAll;

describe('svelte-check-props', () => {
// ...
});
```
(and the same in `angular-check-props.spec.ts`)

File: .eslintignore [hunk 3]
```diff
@@ -7,3 +7,4 @@ test-results/
 .svelte-kit/
 demo/vite-env.d.ts
 demo/src/app.d.ts
+**/environment.prod.ts
```
Comment (divdavem) [inline .eslintignore:10]: Rather than excluding this file, I would prefer adding the file to the list of files in `tsconfig.app.json`.

**Comment by divdavem** [comment]
I have put my suggested changes to this PR in this commit (on my fork of the AgnosUI repository): https://github.com/divdavem/AgnosUI/commit/71f7af28def308ab7097cc1f34769eb3d1bad5b6

File: demo/src/lib/stackblitz/angular/tsconfig.json (hunk 1 above)
Comment (quentinderoubaix) [reply demo/src/lib/stackblitz/angular/tsconfig.json:11]: Thanks for noticing !

File: eslint-plugin/vitest.config.ts (hunk 2 above)
Comment (quentinderoubaix) [reply eslint-plugin/vitest.config.ts]: Great proposal, it has been added thanks to your proposed changes commit.

**Comment by quentinderoubaix** [comment]
Thanks for the review / suggested changes @divdavem !

**Pull request merged**